### DISCUSSION TITLE
maxswerve absolute encoders offsets

### DISCUSSION THREAD

**Discussion by isubsmith** [BroncBotz3481/YAGSL-Example#189] [General]
I noticed Trial-N-Terror has a fork for this, but wondering if there's any thought on official support. I was a bit worried about their last few commit messages in their fork and that it is a ways behind the YAGSL main branch. 

We're using REV maxswerves. The modules come with a hardware jig to hold the wheel at the right orientation per corner and then you reset the absolute encoder offset via the REV client and store that in firmware.  It would be very nice if we could provide a null encoder offset in the YAGSL json files. The team is programmed (haha) to do the zero set via the hw jig if we ever replace a swerve module. I'm concerned we'll forget to change the json. I was looking around the YAGSL code (and the TNT fork) and it looks fairly straightforward but Java is not my native language. I might try to make something work before our next regional (week 5) but official commentary on it would be nice. 

Much appreciated.

**Comment by thenetworkgrinch** [comment]
It is in the backlog of issues we will address but it probably wont happen this season. As of now you will have to set the offsets to 0 in the SparkMAX via the REV Hardware Client with the absolute encoders attached. Set the offset in the JSON based off of the value reported in shuffleboard while wheels are facing forward with bevels to the left (NOT WITH THE JIG!!!) then modify it to set every wheel straight by adding +/-90 to the offsets as needed. Optionally you can call `SwerveDrive.pushOffsetToControllers()` and it will set them in the SparkMAX similar to REV's UI.

**Reply by Technologyman00** [reply]
Honestly I should delete Trial N Terror's repo. (I'm the lead Mentor) I don't think it has what you are looking for besides just a description in the read me of how we did it.

**Comment by isubsmith** [comment]
Hmm, ok, thanks for the advice and comments from both of you. I am not sure we'll switch to YAGSL this season (we have a fully working robot but I was looking forward to some features of YAGSL). I might still give it a few hours this weekend and see how it goes. Thanks!

**Comment by thenetworkgrinch** [comment]
@isubsmith This is fixed in the latest version

**Reply by isubsmith** [reply]
Cool, sounds good. 

We're actually switching to CTRE swerve (X60/X44, CANCoderss, etc) this coming season, but we'll still have the REV environment for our backup and practice bot so I'll give YAGSL another try with both environments.

**Reply by thenetworkgrinch** [reply]
Awesome, I am sure you're going to love it!!
